src/nvim/linematch.c

```diff
@@ -353,24 +362,46 @@ size_t linematch_nbuffers(const char **diff_blk, const int *diff_len, const size
 
   // memory for avoiding repetitive calculations of score
   int df_iters[LN_MAX_BUFS];
-  populate_tensor(df_iters, 0, diffcmppath, diff_len, ndiffs, diff_blk, iwhite);
-
-  const size_t u = unwrap_indexes(diff_len, diff_len, ndiffs);
-  diffcmppath_T *startNode = &diffcmppath[u];
+  bool canceled = 0;
+  populate_tensor(df_iters, 0, diffcmppath, diff_len, ndiffs, diff_blk, iwhite, &canceled);
 
   *decisions = xmalloc(sizeof(int) * memsize_decisions);
   size_t n_optimal = 0;
-  test_charmatch_paths(startNode, 0);
-  while (startNode->df_path_n > 0) {
-    size_t j = startNode->df_optimal_choice;
-    (*decisions)[n_optimal++] = startNode->df_choice[j];
-    startNode = startNode->df_decision[j];
-  }
-  // reverse array
-  for (size_t i = 0; i < (n_optimal / 2); i++) {
-    int tmp = (*decisions)[i];
-    (*decisions)[i] = (*decisions)[n_optimal - 1 - i];
-    (*decisions)[n_optimal - 1 - i] = tmp;
+  if (canceled == true) { // if canceled, compare the lines guessing that the new lines are at the
+                          // bottom of the longer text
```

Comment:
would it simplify things we just bailed out of diff-mode entirely? I don't think we need to bother trying to construct partial "guessed" result.